public/draft/2020-12/draft-bhutton-json-schema-validation-01.html

```diff
@@ -1744,6 +1744,10 @@ <h4 id="name-multipleof">
 <p id="section-6.2.1-2">
                         A numeric instance is valid only if division by this keyword's value results in
                         an integer.<a href="#section-6.2.1-2" class="pilcrow">¶</a></p>
+<p id="section-6.2.1-2">
+                        To ensure consistent validation:
+                        - Use scaled integers (e.g., represent $4.02 as 402 with 'multipleOf: 1')
+                        - If floating-point values are necessary, use binary-friendly numbers (e.g., 0.5, 0.25, 0.125)</p><a href="#section-6.2.1-2" class="pilcrow">¶</a></p>
```

Comment:
Is this really the recommendation we want to make?

Reply:
I'd say "no". Not least because it isn't really meaningful; why would we be recommending that people only use particular values. And then there is the problem of representation on all platforms (e.g. 1.3, 2.4, 5.6 cannot be represented precisely as a 32 bit float)

Reply:
A use case could be defining currency, for example dollars & cents.  You'd want to ensure that your data only has a penny ($0.01) resolution.

But the point is that we don't want to force users into a scenario where they should be trying to figure out the nuances of binary float arithmetic while writing their schemas.

I think that the very most, we should be leaving a note that some implementations may use IEEE7 math.

I do think the way that the spec is worded is pretty cleverly, though.

> 
> [6.2.1.](https://json-schema.org/draft/2020-12/json-schema-validation#section-6.2.1) multipleOf
> 
> The value of "multipleOf" MUST be a number, strictly greater than 0.
> A numeric instance is valid only if division by this keyword's value results in an integer.

In C#, 

```c#
double a = .0075;  // IEEE754
double b = .0001;
decimal c = .0075m;
decimal d = .0001m;

Console.WriteLine("double");
Console.WriteLine(a / b);
Console.WriteLine(a % b);
Console.WriteLine();
Console.WriteLine("decimal");
Console.WriteLine(c / d);
Console.WriteLine(c % d);
```

outputs

```
double
75
9.999999999999937E-05

decimal
75
0.0000
```

So following the method prescribed by the spec (using division instead of modulation) actually produces the correct result, even with IEEE754 math.